power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/common_solver_functions.hpp

```diff
@@ -114,11 +189,12 @@ inline void calculate_pf_result(YBus<sym> const& y_bus, PowerFlowInput<sym> cons
     output.load_gen.resize(load_gens_per_bus.element_size());
     output.bus_injection.resize(sources_per_bus.size());
 
+    output.bus_injection = y_bus.calculate_injection(output.u);
     for (auto const& [bus_number, sources, load_gens] : enumerated_zip_sequence(sources_per_bus, load_gens_per_bus)) {
-        calculate_source_result<sym>(sources, bus_number, y_bus, input, output);
-        calculate_load_gen_result<sym>(load_gens, bus_number, input, output, load_gen_func);
+        ComplexValue<sym> i_load_gen_bus{};
+        calculate_load_gen_result<sym>(load_gens, bus_number, input, output, load_gen_func, i_load_gen_bus);
+        calculate_source_result<sym>(sources, bus_number, y_bus, input, output, i_load_gen_bus);
```

Comment:
Instead, first check if there are sources at the bus at all.

If yes, then calculate the `i_load_gen_bus`. If not just continue to the next bus.